script/validate-wasm-grammar-prs.js

```diff
@@ -0,0 +1,148 @@
+      // that this particular one is using the tree-sitter wasm file that was
+      // actually changed.
+      const grammarFile = contents.treeSitter?.grammar ?? "";
+
+      if (path.basename(grammarFile) === path.basename(wasmFile)) {
+        // This grammar uses the WASM file that's changed. So we must ensure our key has also changed
+        // Sidenote we use `basename` here, since the `wasmFile` will be
+        // a path relative from the root of the repo, meanwhile `grammarFile`
+        // will be relative from the file itself
+
+        // In order to check the previous state of what the key is, we first must retreive the file prior to this PR
+        const getPrevFile = cp.spawnSync("git", [ "show", `${commit}:./${filePath}` ]);
+
+        if (getPrevFile.status !== 0 || getPrevFile.stderr.toString().length > 0) {
+          // This can fail for two major reasons
+          // 1. The `git show` command has returned an error code other than `0`, failing.
+          // 2. This is a new file, and it failed to find an earlier copy (which didn't exist)
+          // So that we don't fail brand new TreeSitter grammars, we manually check for number 2
+
+          if (getPrevFile.stderr.toString().includes("exists on disk, but not in")) {
+            // Looks like this file is new. Skip this check
+            if (verbose) {
+              console.log("Looks like this file is new. Skipping...");
+            }
+            continue;
+          }
+
+          console.error("Git command failed!");
+          console.error(`'git show ${commit}:./${filePath}'`);
+          console.error(getPrevFile.stderr.toString());
+          process.exit(1);
+        }
+
+        fs.writeFileSync(path.join(wasmPath, "..", `OLD-${file}`), getPrevFile.stdout.toString());
+
+        const oldContents = CSON.readFileSync(path.join(wasmPath, "..", `OLD-${file}`));
+        const oldParserSource = oldContents.treeSitter?.parserSource ?? "";
+        const newParserSource = contents.treeSitter?.parserSource ?? "";
+
+        if (newParserSource.length === 0) {
+          console.error(`Failed to find the new \`parserSource\` within: '${filePath}'`);
+          console.error(contents.treeSitter);
+          process.exit(1);
+        }
+
+        if (oldParserSource == newParserSource) {
+          // The repo and commit is identical! This means it hasn't been updated
+          console.error(`The \`parserSource\` key of '${filePath}' has not been updated!`);
+          console.error(`Current key: ${newParserSource} - Old key: ${oldParserSource}`);
+          process.exit(1);
+        }
+
+        // Else it looks like it has been updated properly
+        console.log(`Validated \`parserSource\` has been updated within '${filePath}' properly.`);
+      } else {
+        if (verbose) {
+          console.log("This grammar file doesn't use a WASM file that's changed (On the current interation)");
```

Comment:
```suggestion
          console.log("This grammar file doesn't use a WASM file that's changed (On the current iteration)");
```

Small typo, I think this is meant to be "iteration"?

Reply:
Lol thanks for catching it